### DISCUSSION TITLE
Does backtesting always conclude by closing the last trade? | How to see open trades?

### DISCUSSION THREAD

**Discussion by cgdutoit** [kernc/backtesting.py#933] [Q&A]
Hi - After a strategy is finished being backtested, I wanted to find out if we're currently in an open trade for this strategy.

Let's say I run a simple moving average crossover strategy such as:

<img width="571" alt="Screenshot 2023-03-12 at 11 27 41 AM" src="https://user-images.githubusercontent.com/9360406/224561741-26ed3fad-1d54-4c6c-95ac-d2a7f383b317.png">

At the conclusion of the backtest, I'd like to know if we currently have an open position (ie. the exit criteria havent been triggered as of the end of my dataset).

When I query stats["_trades"] it always shows the last trade as "exited". If you see this image below, you can see that it exited the trade on 3/10/23 which is the last day of my dataset, but that trade, by the strategy conditions, should still be open.

<img width="704" alt="Screenshot 2023-03-12 at 11 22 25 AM" src="https://user-images.githubusercontent.com/9360406/224561836-74fa7051-165c-4c83-9658-a317e4a436bc.png">

So my two questions are:

1. Does backtesting always exit the trade on the last day of your dataset regardless if the exit criteria haven't been met?
2. How would I, at the conclusion of the backtest, know if a current open trade exists for that strategy? ie. In the above code where I have `y=stats["_trades"]` what would I use to see if an open trade currently exists?

Thanks for any insight

**Comment by aliencaocao** [comment]
> Does backtesting always exit the trade on the last day of your dataset regardless if the exit criteria haven't been met?

Yes and this is by design

If you want to see open trades that should not be closed, just look for those closed on the last bar
